Adding src/test/java with compile failures causes "Unable to execute mojo" stacktrace

1. Start multi module dev mode on https://github.com/turkeylurkey/lmp-multimodules/tree/main/testCases/typeB
2. Copy tests from https://github.com/turkeylurkey/lmp-multimodules/tree/main/testCases/typeA/jar/src/test/java/io/openliberty/guides/multimodules/lib into the typeB project's `jar/src` directory.  Compilation failure is expected because pom.xml does not have the proper dependencies.  But the following stacktrace also occurs which may look alarming/confusing.

```
[INFO] Compiling 1 source file to C:\Users\Eric\git\lmp-multimodules\testCases\typeB\jar\target\test-classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /C:/Users/Eric/git/lmp-multimodules/testCases/typeB/jar/src/test/java/io/openliberty/guides/multimodules/lib/ConverterUnitTest.java:[3,29] package org.junit.jupiter.api does not exist
[ERROR] /C:/Users/Eric/git/lmp-multimodules/testCases/typeB/jar/src/test/java/io/openliberty/guides/multimodules/lib/ConverterUnitTest.java:[5,36] package org.junit.jupiter.api does not exist
[ERROR] /C:/Users/Eric/git/lmp-multimodules/testCases/typeB/jar/src/test/java/io/openliberty/guides/multimodules/lib/ConverterUnitTest.java:[5,1] static import only from classes and interfaces
[ERROR] /C:/Users/Eric/git/lmp-multimodules/testCases/typeB/jar/src/test/java/io/openliberty/guides/multimodules/lib/ConverterUnitTest.java:[9,6] cannot find symbol
  symbol:   class Test
  location: class io.openliberty.guides.multimodules.lib.ConverterUnitTest
[ERROR] /C:/Users/Eric/git/lmp-multimodules/testCases/typeB/jar/src/test/java/io/openliberty/guides/multimodules/lib/ConverterUnitTest.java:[12,9] cannot find symbol
  symbol:   method assertEquals(int,int)
  location: class io.openliberty.guides.multimodules.lib.ConverterUnitTest
[INFO] 5 errors
[INFO] -------------------------------------------------------------
[ERROR] Unable to compile
org.apache.maven.plugin.MojoExecutionException: Unable to execute mojo
    at org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo (MojoExecutor.java:132)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runMojoForProject (StartDebugMojoSupport.java:206)
    at io.openliberty.tools.maven.server.DevMojo$DevMojoUtil.compile (DevMojo.java:743)
    at io.openliberty.tools.common.plugins.util.DevUtil.watchFiles (DevUtil.java:2732)
    at io.openliberty.tools.maven.server.DevMojo.doExecute (DevMojo.java:1029)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:858)
    at org.apache.maven.plugin.compiler.TestCompilerMojo.execute (TestCompilerMojo.java:152)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo (MojoExecutor.java:119)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runMojoForProject (StartDebugMojoSupport.java:206)
    at io.openliberty.tools.maven.server.DevMojo$DevMojoUtil.compile (DevMojo.java:743)
    at io.openliberty.tools.common.plugins.util.DevUtil.watchFiles (DevUtil.java:2732)
    at io.openliberty.tools.maven.server.DevMojo.doExecute (DevMojo.java:1029)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
@@ -1340,17 +1340,11 @@ public class DevMojo extends StartDebugMojoSupport {
      *                     default project is to be used
      * @throws MojoExecutionException
      */
-    private void runCompileMojo(String goal, MavenProject currentProject) throws MojoExecutionException {
-        Plugin plugin;
+    private void runCompileMojo(String goal, MavenProject mavenProject) throws MojoExecutionException {
+        Plugin plugin = getPluginForProject("org.apache.maven.plugins", "maven-compiler-plugin", mavenProject);
         MavenSession tempSession = session.clone();
-        MavenProject tempProject = project;
-        if (currentProject != null) {
-            plugin = getPluginForProject("org.apache.maven.plugins", "maven-compiler-plugin", currentProject);
-            tempSession.setCurrentProject(currentProject);
-            tempProject = currentProject;
-        } else {
-            plugin = getPlugin("org.apache.maven.plugins", "maven-compiler-plugin");
-        }
+        tempSession.setCurrentProject(mavenProject);
+        MavenProject tempProject = mavenProject;
         Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(plugin, goal, log);
         config = Xpp3Dom.mergeXpp3Dom(configuration(element(name("failOnError"), "false")), config);
         log.info("Running maven-compiler-plugin:" + goal + " on " + tempProject.getFile());
@@ -1364,7 +1358,7 @@ public class DevMojo extends StartDebugMojoSupport {
      * @throws MojoExecutionException
      */
     private void runCompileMojoLogWarning() throws MojoExecutionException {
-        runCompileMojo("compile", null);
+        runCompileMojo("compile", project);
     }
 
     /**
@@ -1382,7 +1376,7 @@ public class DevMojo extends StartDebugMojoSupport {
      * @throws MojoExecutionException
      */
     private void runTestCompileMojoLogWarning() throws MojoExecutionException {
-        runCompileMojo("testCompile", null);
+        runCompileMojo("testCompile", project);
     }
 
     /**

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
@@ -717,11 +717,11 @@ public class DevMojo extends StartDebugMojoSupport {
         public boolean compile(File dir) {
             try {
                 if (dir.equals(sourceDirectory)) {
-                    runMojo("org.apache.maven.plugins", "maven-compiler-plugin", "compile");
+                    runCompileMojoLogWarning();
                     runMojo("org.apache.maven.plugins", "maven-resources-plugin", "resources");
                 }
                 if (dir.equals(testSourceDirectory)) {
-                    runMojo("org.apache.maven.plugins", "maven-compiler-plugin", "testCompile");
+                    runTestCompileMojoLogWarning();
                     runMojo("org.apache.maven.plugins", "maven-resources-plugin", "testResources");
                 }
                 return true;
@@ -733,15 +733,15 @@ public class DevMojo extends StartDebugMojoSupport {
 
         @Override
         public boolean compile(File dir, ProjectModule project) {
-            MavenProject mavenProj = resolveMavenProject(project.getBuildFile());
+            MavenProject mavenProject = resolveMavenProject(project.getBuildFile());
             try {
                 if (dir.equals(project.getSourceDirectory())) {
-                    runMojoForProject("org.apache.maven.plugins", "maven-compiler-plugin", "compile", mavenProj);
-                    runMojoForProject("org.apache.maven.plugins", "maven-resources-plugin", "resources", mavenProj);
+                    runCompileMojoLogWarning(mavenProject);
+                    runMojoForProject("org.apache.maven.plugins", "maven-resources-plugin", "resources", mavenProject);
                 }
                 if (dir.equals(project.getTestSourceDirectory())) {
-                    runMojoForProject("org.apache.maven.plugins", "maven-compiler-plugin", "testCompile", mavenProj);
-                    runMojoForProject("org.apache.maven.plugins", "maven-resources-plugin", "testResources", mavenProj);
+                    runTestCompileMojoLogWarning(mavenProject);
+                    runMojoForProject("org.apache.maven.plugins", "maven-resources-plugin", "testResources", mavenProject);
                 }
                 return true;
             } catch (MojoExecutionException e) {
@@ -1335,16 +1335,27 @@ public class DevMojo extends StartDebugMojoSupport {
      * Executes Maven goal passed but sets failOnError to false All errors are
      * logged as warning messages
      * 
-     * @param goal Maven compile goal
+     * @param goal         Maven compile goal
+     * @param MavenProject Maven project to run compile goal against, null if
+     *                     default project is to be used
      * @throws MojoExecutionException
      */
-    private void runCompileMojo(String goal) throws MojoExecutionException {
-        Plugin plugin = getPlugin("org.apache.maven.plugins", "maven-compiler-plugin");
+    private void runCompileMojo(String goal, MavenProject currentProject) throws MojoExecutionException {
+        Plugin plugin;
+        MavenSession tempSession = session.clone();
+        MavenProject tempProject = project;
+        if (currentProject != null) {
+            plugin = getPluginForProject("org.apache.maven.plugins", "maven-compiler-plugin", currentProject);
+            tempSession.setCurrentProject(currentProject);
+            tempProject = currentProject;
+        } else {
+            plugin = getPlugin("org.apache.maven.plugins", "maven-compiler-plugin");
+        }
         Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(plugin, goal, log);
         config = Xpp3Dom.mergeXpp3Dom(configuration(element(name("failOnError"), "false")), config);
-        log.info("Running maven-compiler-plugin:" + goal);
+        log.info("Running maven-compiler-plugin:" + goal + " on " + tempProject.getFile());
         log.debug("configuration:\n" + config);
-        executeMojo(plugin, goal(goal), config, executionEnvironment(project, session, pluginManager));
+        executeMojo(plugin, goal(goal), config, executionEnvironment(tempProject, tempSession, pluginManager));
     }
 
     /**
@@ -1353,7 +1364,16 @@ public class DevMojo extends StartDebugMojoSupport {
      * @throws MojoExecutionException
      */
     private void runCompileMojoLogWarning() throws MojoExecutionException {
-        runCompileMojo("compile");
+        runCompileMojo("compile", null);
+    }
+
+    /**
+     * Executes maven:compile but logs errors as warning messages
+     * 
+     * @throws MojoExecutionException
+     */
+    private void runCompileMojoLogWarning(MavenProject mavenProject) throws MojoExecutionException {
+        runCompileMojo("compile", mavenProject);
     }
 
     /**
@@ -1362,7 +1382,16 @@ public class DevMojo extends StartDebugMojoSupport {
      * @throws MojoExecutionException
      */
     private void runTestCompileMojoLogWarning() throws MojoExecutionException {
-        runCompileMojo("testCompile");
+        runCompileMojo("testCompile", null);
+    }
+
+    /**
+     * Executes maven:testCompile but logs errors as warning messages
+     * 
+     * @throws MojoExecutionException
+     */
+    private void runTestCompileMojoLogWarning(MavenProject mavenProject) throws MojoExecutionException {
+        runCompileMojo("testCompile", mavenProject);
     }
 
     /**

